File: 392-is-subsequence/392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        temp = 0

        for c in s:

            i = temp

            while i < len(t):
                if t[i] == c:
                    break
                i += 1

            if i == len(t):
                return False

            temp = i+1

        return True

File: 392-is-subsequence/392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        
        if(t.length() < s.length())
            return false;
        
        if(t.length() == s.length())
            return t.equals(s);
        
        int j = 0;
        
        int charS = s.length();
        int charT = t.length();
        
        for(int i=0;i<t.length() && j<s.length();i++){
            
            if(s.charAt(j) == t.charAt(i)){
                j++;
                charS--;
                charT--;
                continue;
            }
            
                charT--;
            
            if(charS > charT)
                return false;
            
        }
        
        return true;
        
    }
}